client/GameComponents/Card.jsx

```diff
@@ -32,7 +32,7 @@ class Card extends React.Component {
     }
 
     isAllowedMenuSource() {
-        return this.props.source === 'play area' || this.props.source === 'agenda' || this.props.source === 'revealed plots';
+        return this.props.source === 'play area' || this.props.source === 'agenda' || this.props.source === 'revealed plots' || _.any(this.props.card.menu, action => action.canPlayFrom === this.props.source) ;
```

Comment:
Yes... this was a half measure that worked for the specific case of lady in waiting and not more generally.  Thinking on this further, it might be necessary to default the canPlayFrom properly ('play area' for draw cards, 'revealed plots' for plots, and 'agenda' for agendas) and then change this to be based entirely off canPlayFrom.